api/napi/src/interpreter/component_instance.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-1.1 OR LicenseRef-Slint-commercial
+
+use napi::{Env, Error, JsFunction, JsUnknown, NapiRaw, NapiValue, Ref, Result};
+use slint_interpreter::{ComponentHandle, ComponentInstance, Value};
+
+use super::JsComponentDefinition;
+
+#[napi(js_name = "JsComponentInstance")]
+pub struct JsComponentInstance {
+    inner: ComponentInstance,
+}
+
+impl From<ComponentInstance> for JsComponentInstance {
+    fn from(instance: ComponentInstance) -> Self {
+        Self { inner: instance }
+    }
+}
+
+#[napi]
+impl JsComponentInstance {
+    #[napi(constructor)]
+    pub fn new() -> Self {
+        unreachable!("ComponentDefinition can only be created by using ComponentCompiler.")
```

Comment:
Is panicking really the right thing to do in this case?  Shouldn't that be a javascript exception? Or are panics transformed into exceptions?

Reply:
I think if there is a need to keep this constructor, then I think that it should thrown an assertion in JavaScript, instead of aborting the process. Please either remove it (if possible) or if it needs to be kept, please add a test that verifies that calling this constructor throws an exception as expected.

(the same applies also to `ComponentDefinition`)